File: backend/models/listing.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ListingItem(BaseModel):
    id: str
    slug: str
    item_type: str  # "section" or "content"
    content_type: Optional[str] = None
    title: str
    summary: Optional[str] = None
    image_url: Optional[str] = None
    video_url: Optional[str] = None
    path: Optional[str] = None
    display_type: Optional[str] = None
    tags: list[str] = []
    is_published: bool = True
    is_featured: bool = False
    sort_order: Optional[int] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    user_id: Optional[str] = None

File: frontend/src/shared/types/listing.ts
export interface ListingItem {
    id: string;
    slug: string;
    item_type: 'section' | 'content';
    content_type?: string;
    title: string;
    summary?: string | null;
    image_url?: string | null;
    video_url?: string | null;
    path?: string | null;
    display_type?: string | null;
    tags: string[];
    is_published: boolean;
    is_featured: boolean;
    sort_order?: number | null;
    created_at?: string | null;
    updated_at?: string | null;
    user_id?: string | null;
}
